### ISSUE TITLE
Bug: either max or maximum option should be allowed in max-params

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#20322]
### Environment

playground

### What parser are you using?

Default (Espree)

### What did you do?

```js
/*eslint max-params: ["error", { "max": 2, "maximum": 3 }]*/

function hasNoThis(first, second, third) {
	// ...
}
```

### What did you expect to happen?

validation error, either max or maximum can be specified.

### What actually happened?

maximum takes priority

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbWF4LXBhcmFtczogW1wiZXJyb3JcIiwgeyBcIm1heFwiOiAyLCBcIm1heGltdW1cIjogMyB9XSovXG5cbmZ1bmN0aW9uIGhhc05vVGhpcyhmaXJzdCwgc2Vjb25kLCB0aGlyZCkge1xuXHQvLyAuLi5cbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlciI6IkB0eXBlc2NyaXB0LWVzbGludC9wYXJzZXIiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9LCJzb3VyY2VUeXBlIjoibW9kdWxlIn19fX0=

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

We can also consider removing the maximum option as a breaking change

**Comment by KirtiGautam620** [comment]
Hi! I’m a beginner and I’d like to contribute to this issue. 
Could you please assign it to me? I want to add a comment in the code and update the documentation to clarify that "maximum" takes priority over "max".

**Comment by mdjermanovic** [comment]
> validation error, either max or maximum can be specified.

Per our [semver policies](https://github.com/eslint/eslint?tab=readme-ov-file#semantic-versioning-policy), this would be a breaking change.

> - Major release (likely to break your lint build)
>     - `eslint:recommended` is updated and may result in new linting errors (e.g., rule additions, most rule option updates).
>     - A new option to an existing rule that results in ESLint reporting more linting errors by default.
>     - An existing formatter is removed.
>     - **Part of the public API is removed or changed in an incompatible way. The public API includes**:
>         - **Rule schemas**
>         - Configuration schema
>         - Command-line options
>         - Node.js API
>         - Rule, formatter, parser, plugin APIs

Since the scope for v10 is frozen, I'm adding this to v11.0.0 - Need Discussion.

**Comment by DMartens** [comment]
The same problem also exists for other core rules which should then also be updated:
- `complexity`
- `max-depth`
- `max-nested-callbacks`
- `max-statements`

**Comment by mdjermanovic** [comment]
The `maximum` options in these 5 rules have been documented as deprecated for 10 years, so it might be a good time to remove them, as mentioned in https://github.com/eslint/eslint/issues/5685.

**Comment by ivangray271-lgtm** [comment]
I agree that since the maximum options for these rules have been deprecated
for 10 years, now would be a good time to move forward with removing them
as discussed in #5685.
